pkg/apis/networking/v1/types.go

```diff
@@ -174,3 +175,144 @@ type NetworkList struct {
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +k8s:openapi-gen=true
+
+// RemoteCluster is a specification for a RemoteCluster resource
+type RemoteCluster struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   RemoteClusterSpec   `json:"spec"`
+	Status RemoteClusterStatus `json:"status"`
+}
+
+// RemoteClusterSpec is the spec for a RemoteCluster resource
+type RemoteClusterSpec struct {
+	ConnConfig APIServerConnConfig `json:"connConfig"`
+}
+
+type APIServerConnConfig struct {
+	// apiServer address. Format: https://ip:port
+	Endpoint   string `json:"endpoint"`
+	CABundle   []byte `json:"caBundle"`
+	ClientCert []byte `json:"clientCert"`
+	ClientKey  []byte `json:"clientKey"`
+	// The maximum length of time to wait before giving up on a server
+	// request. A value of zero means no timeout. Default: zero second
+	Timeout uint32 `json:"timeout"`
+}
+
+// RemoteClusterStatus is the status for a RemoteCluster resource
+type RemoteClusterStatus struct {
+	// Conditions is an array of current cluster conditions.
+	Conditions []ClusterCondition `json:"conditions"`
+}
+
+// ClusterCondition describes current state of a cluster.
+type ClusterCondition struct {
+	// Type of cluster condition, Ready or Offline.
+	Type ClusterConditionType `json:"type"`
+	// Status of the condition, one of True, False, Unknown.
+	Status apiv1.ConditionStatus `json:"status"`
+	// Last time the condition was checked.
+	LastProbeTime metav1.Time `json:"lastProbeTime"`
+	// Last time the condition transit from one status to another.
+	// +optional
+	LastTransitionTime *metav1.Time `json:"lastTransitionTime,omitempty"`
+	// (brief) reason for the condition's last transition.
+	// +optional
+	Reason *string `json:"reason,omitempty"`
+	// Human readable message indicating details about last transition.
+	// +optional
+	Message *string `json:"message,omitempty"`
+}
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// RemoteClusterList is a list of RemoteCluster resources
+type RemoteClusterList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata"`
+
+	Items []RemoteCluster `json:"items"`
+}
+
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +k8s:openapi-gen=true
+
+// RemoteSubnet is a specification for a RemoteSubnet resource
+type RemoteSubnet struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   RemoteSubnetSpec   `json:"spec"`
+	Status RemoteSubnetStatus `json:"status"`
+}
+
+// RemoteSubnetSpec is the spec for a RemoteSubnet resource
+type RemoteSubnetSpec struct {
+	Version      IPVersion   `json:"version"`
+	CIDR         string      `json:"cidr"`
+	Type         NetworkType `json:"type"`
+	ClusterName  string      `json:"clusterName"`
+	OverlayNetID *uint32     `json:"overlayNetID,omitempty"`
```

Comment:
If underlay network will also have a RemoteSubnet CR, maybe 'NetID' is better than 'OverlayNetID'.

Reply:
First, this is pointer, if it is underlay network, it is nil and will not be printed.
Second, this is actually a vxlan network identifier in all connected domain, I this OverlayNetID is more clear @hhyasdf

Reply:
If it is actually a vxlan network identifier in all connected domain, I think maybe it's a cluster-scoped information, why not keep it in RemoteCluster CR?

Reply:
Sound good to me